tmVhdlProducer/vhdlhelper.py

```diff
@@ -355,7 +383,10 @@ def __init__(self, collection):
         self.scale_set = eventSetup.getScaleSetName()
         self.version = VersionHelper(tmEventSetup.__version__)
         self.sw_version = VersionHelper(__version__)
-
+        # hash value of all .py and .vhd files in directory "tmVhdlProducer" recursively
+        file_path = os.path.dirname(__file__)
+        self.files_hash_value = get_files_hash_value(file_path)
```

Comment:
Try to avoid calculating the hash multiple times, this gives potential for discrepancies and introduces an overhead. Try to calculate the hash once in `__main__` and pass it like the `__version__` variable.

Reply:
done